File: pythonwars/BorrowerSpeak.py
# https://www.codewars.com/kata/57d2ba8095497e484e00002e/train/python


def borrow(s):
    punctuation = "!@#$%^&*()_+<>?:.,;"
    no_punct = " "  # create empty string - this to push new string into - the result
    for c in s:  # checking each character in string
        # if character is not a punctuation then we create no_punct and add no_punct to characters (c)
        if c not in punctuation:
            no_punct = no_punct + c
            lower = no_punct.lower()  # then we set our new string all to lowercase
            # finally we remove the space between the words and return ONE string
            space = lower.replace(' ', '')
    return (space)


print(borrow('WhAt! FiCK! DaMn CAke?'))


# trial 2 - almost!! just need to delete the space
# def borrow(s):
#    punctuation = "!@#$%^&*()_+<>?:.,;"
#    no_punct = " "
#    for c in s:
#        if c not in punctuation:
#            no_punct = no_punct + c
#            lower = no_punct.lower()
#    return (lower)

# trial 1 - delete all upperCase
# def borrow(s):
#    new = s.lowercase
#    return new

# delete capitals
# delete punctuations
# return one string - no space
# string.lower() - converts string to lowercase

File: BorrowerSpeak.js
// https://www.codewars.com/kata/57d2ba8095497e484e00002e/train/javascript

function borrow(s) {
  splitStr = s.split('')
  let word = splitStr.filter(function (val) { //making each individual letter a part of array like = ["W","h","A","t"]
    return val !== " " && val !== "!" && val !== "?" && val !== "." && val !== "," && val !== "_" && val !== ":" && val !== ";"
  })  // then do NOT return punctutations 
  return word.join("").toLowerCase()
}

console.log(borrow('WhAt! FiCK! DaMn CAke?'))

// if punct, delete puctution to empty ''
//any empty ' '  - make sure it becomes one string
// if capital toLowerCase

//Below support on methods:
// The split() method divides a String into an ordered list of substrings, puts these substrings into an array, and returns the array.  
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split
//The join() method creates and returns a new string by concatenating all of the elements in an array (or an array-like object),

// trial
// function borrow(s) {
//   punct = "!@#$%^&*()_+<>?:.,;"
//   newS = " "
//   if (s !== punct) {
//     return s
//   }
//   return
// }